circle.yml

```diff
@@ -516,10 +523,6 @@ jobs:
       - checkout
       - build-release-core-1_0
       - generate-documentation
-      #     Uncomment when API' docs will be ready
-      #     - run:
-      #         name: Check public documentation
-      #         command: sh scripts/dokka-validate.sh
```

Comment:
Do we want to `check-public-documentation` here instead of only doing it as part of `build-1_0`? 🤔 If so, it'd be something like this

```
- build-release-core-1_0
- check-public-documentation
- generate-documentation
```

What do you think?

Same would apply to the rest of the `jobs` 👇

Reply:
yeah, I'm thinking that it should be a responsibility of developer that bring a new API. Documentation should stay cleaner. 
Just image situation when you(or somebody else) are gonna make new QA or even Release build, but you cannot because dokka fails, and you aren't so familiar with API that is not covered by doc. Dev who wrote this API know this API better than somebody else. @Guardiola31337

Reply:
Totally agree. I wasn't saying removing the one from `build-1_0` (sorry if I didn't make myself clear) but adding this "extra" check to the release `jobs`. What do you think? Wondering if something could happen in between PRs and releases that this additional check can catch 🤔

Reply:
@Guardiola31337 oh, I see. I'm trying to figure out how it could happen that undocumented code appear in `master` when validation script is enabled. Could you provide a sample? 🤔

Reply:
Shouldn't be a problem in most of the cases but wondering if something could slip in (when merging PRs that are not rebased from latest changes in `master` for example, saw some conflicts in the past because of that https://github.com/mapbox/mapbox-navigation-android/pull/2720#issuecomment-610502285) causing documentation issues when running the release jobs.

Reply:
yep, in this case we just get invalid state in `master` and fix it **before** release 😉 
anyway, I've added checks for release builds https://github.com/mapbox/mapbox-navigation-android/pull/2761/commits/72b40d3124ee1273ecd3dc9bb55d89dea580add5

Reply:
Yeah, that's correct but we're not being notified so we can miss it 😅 This will be like a safety net catching regressions before releasing 😉

Reply:
I believe we should add it to the `release-ui-1_0` jobs too. What do you think?

cc @abhishek1508

Reply:
ui is fully driven by java, not kotlin, right? so no sense

Reply:
There are some Kotlin 😅 and will be more and more so good to have it 😉

Reply:
so we must set-up dokka for UI 😃

Reply:
cc @abhishek1508

Reply:
dokka for ui is already setup @RingerJK

Reply:
@abhishek1508 

Capturing from @RingerJK 
> it’s plugin, not config like https://github.com/mapbox/mapbox-navigation-android/blob/master/libnavigation-core/build.gradle#L9